kubernetes-client/src/test/java/io/fabric8/kubernetes/client/behavior/RequestTimeoutTest.java

```diff
@@ -0,0 +1,274 @@
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.fabric8.kubernetes.client.RequestConfigBuilder;
+import io.fabric8.kubernetes.client.dsl.LogWatch;
+import io.fabric8.kubernetes.client.http.AsyncBody;
+import io.fabric8.kubernetes.client.http.TestAsyncBody;
+import io.fabric8.kubernetes.client.http.TestHttpResponse;
+import io.fabric8.kubernetes.client.http.TestStandardHttpClient;
+import io.fabric8.kubernetes.client.http.TestStandardHttpClientFactory;
+import io.fabric8.kubernetes.client.http.WebSocket;
+import io.fabric8.kubernetes.client.http.WebSocketResponse;
+import io.fabric8.kubernetes.client.http.WebSocketUpgradeResponse;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@SuppressWarnings("resource")
+@DisplayName("Request Timeout Behavior")
+class RequestTimeoutTest {
+
+  private TestStandardHttpClientFactory factory;
+  private KubernetesClient client;
+
+  @BeforeEach
+  void setUp() {
+    factory = new TestStandardHttpClientFactory();
+    client = new KubernetesClientBuilder().withHttpClientFactory(factory).build();
+  }
+
+  @DisplayName("Standard Kubernetes Object operations have default timeout")
+  @Test
+  void standardOperationsHaveDefaultTimeout() {
+    // Given
+    factory.getInstances().iterator().next()
+        .expect("/api/v1/namespaces/.+/pods.+", new TestHttpResponse<AsyncBody>().withCode(200).withBody(new TestAsyncBody()));
+    // When
+    client.pods().withName("foo").get();
+    // Then
+    assertThat(factory.getInstances())
+        .singleElement()
+        .extracting(TestStandardHttpClient::getRecordedConsumeBytesDirects)
+        .asList()
+        .hasSize(1)
+        .singleElement()
+        .extracting("request")
+        .extracting("timeout")
+        .asInstanceOf(InstanceOfAssertFactories.DURATION)
+        .hasSeconds(10);
+  }
+
+  @DisplayName("Standard Kubernetes Object operations, with overridden config, have overridden timeout")
+  @Test
+  void standardOperationsHaveOverriddenTimeoutInConfig() {
+    // Given
+    factory.getInstances().iterator().next()
+        .expect("/api/v1/namespaces/.+/pods.+", new TestHttpResponse<AsyncBody>().withCode(200).withBody(new TestAsyncBody()));
+    // When
+    client.getConfiguration().getRequestConfig().setRequestTimeout(1337000);
```

Comment:
Directly modifying the RequestConfig held by the Config may not work in perpetuity - ideally we're moving toward having the Config and stuff under it as immutable.  At the very least this shouldn't be considered a supported path.

Reply:
Shouldn't be supported but it's possible now.
I'll add a note to clarify that this is not something intended to be used by users.